lib/Graph/Graph.cpp

```diff
@@ -1907,9 +1907,31 @@ void Function::verify() const {
     }
   }
 
+  std::unordered_map<const Variable *, const Node *> variablesWrittenTo;
   for (const auto &N : nodes_) {
     assert(N.getParent() == this &&
            "Node is not linked to the function it belongs");
     N.verify();
+    // Make sure all the variables are at most written once.
+    for (size_t idx = 0, e = N.getNumInputs(); idx < e; ++idx) {
+      auto &input = N.getNthInput(idx);
+      if (!isa<Variable>(input)) {
+        continue;
+      }
+      if (!N.isOverwrittenNthInput(idx)) {
```

Comment:
This check should be the first. If it fails, we don't care of the input is a variable or not.

Reply:
That's true the other way around too.

Reply:
My point was that `N.getNthInput(idx)` and `isa<Variable>(input)` are probably more expensive than `isOverwrittenNthInput`. 
But on the other hand, it is a verifier, so who cares ;-)